stac_fastapi/eodag/extensions/data_download.py

```diff
@@ -181,6 +182,18 @@ def get_data(
                     raise NotFoundError(f"Item {item_id} does not exist. Please order it first") from e
                 raise NotFoundError(e) from e
 
+        if product.downloader_auth and asset_name and asset_name != "downloadLink":
+            asset_values = product.assets[asset_name]
+            # return presigned url if available
+            try:
+                presigned_url = product.downloader_auth.presign_url(asset_values)
+                headers = {"content-disposition": f"attachment; filename={asset_name}"}
```

Comment:
I believe headers such as `content-disposition` are pointless when returning a `302` HTTP response. The filename will come from the next request.